helios-services/src/test/java/com/spotify/helios/servicescommon/ZooKeeperRegistrarUtilTest.java

```diff
@@ -0,0 +1,129 @@
+
+  @Test
+  public void testRegisterHost() throws Exception {
+    final String idPath = Paths.configHostId(HOSTNAME);
+    verify(zkClient).ensurePath(Paths.configHost(HOSTNAME));
+    verify(zkClient).ensurePath(Paths.configHostJobs(HOSTNAME));
+    verify(zkClient).ensurePath(Paths.configHostPorts(HOSTNAME));
+    verify(zkClient).ensurePath(Paths.statusHost(HOSTNAME));
+    verify(zkClient).ensurePath(Paths.statusHostJobs(HOSTNAME));
+    verify(zkClient).createAndSetData(idPath, ID.getBytes(UTF_8));
+    ZooKeeperRegistrarUtil.registerHost(zkClient, idPath, HOSTNAME, ID);
+  }
+
+  @Test
+  public void testDeregisterHost() throws Exception {
+    when(zkClient.getChildren(Paths.statusHostJobs(HOSTNAME))).thenReturn(JOB_STRINGS);
+    when(zkClient.listRecursive(Paths.configHostJob(HOSTNAME, JOB_ID1))).thenReturn(
+        ImmutableList.of("zk-node1"));
+    when(zkClient.exists(Paths.configJobHost(JOB_ID1, HOSTNAME))).thenReturn(new Stat());
+    when(zkClient.listRecursive(Paths.historyJobHost(JOB_ID1, HOSTNAME))).thenReturn(
+        ImmutableList.of("history1", "history2"));
+    when(zkClient.listRecursive(Paths.statusHost(HOSTNAME))).thenReturn(
+        ImmutableList.of("status1", "status2"));
+    when(zkClient.getChildren(Paths.configHostPorts(HOSTNAME))).thenReturn(
+        ImmutableList.of("1", "2"));
+    when(zkClient.exists(Paths.configHostId(HOSTNAME))).thenReturn(new Stat());
+
+    final List<ZooKeeperOperation> ops = Lists.newArrayList();
+
+    ZooKeeperRegistrarUtil.deregisterHost(zkClient, HOSTNAME);
+
+    verify(zkClient).getChildren(Paths.statusHostJobs(HOSTNAME));
+
+    for (final JobId jobId : JOB_IDS) {
+      // Remove host status for job
+      verify(zkClient).listRecursive(Paths.configHostJob(HOSTNAME, jobId));
+      // Remove job status on host
+      verify(zkClient).exists(Paths.configJobHost(jobId, HOSTNAME));
+      // Remove history for each job
+      verify(zkClient).listRecursive(Paths.historyJobHost(jobId, HOSTNAME));
+    }
+
+    ops.add(new Delete("zk-node1"));
+    ops.add(new Delete(Paths.configJobHost(JOB_ID1, HOSTNAME)));
+    ops.add(new Delete("history2"));
+    ops.add(new Delete("history1"));
+    ops.add(new Delete(Paths.configHostJobs(HOSTNAME)));
+    ops.add(new Delete("status2"));
+    ops.add(new Delete("status1"));
+
+    // Remove host status
+    verify(zkClient).listRecursive(Paths.statusHost(HOSTNAME));
+
+    // Remove port allocations
+    verify(zkClient).getChildren(Paths.configHostPorts(HOSTNAME));
+    ops.add(new Delete(Paths.configHostPort(HOSTNAME, 1)));
+    ops.add(new Delete(Paths.configHostPort(HOSTNAME, 2)));
+    ops.add(new Delete(Paths.configHostPorts(HOSTNAME)));
+
+    // Remove host id
+    verify(zkClient).exists(Paths.configHostId(HOSTNAME));
+    ops.add(new Delete(Paths.configHostId(HOSTNAME)));
+
+    // Remove host config root
+    ops.add(new Delete(Paths.configHost(HOSTNAME)));
+
+    verify(zkClient).transaction(ops);
```

Comment:
@rohansingh Should we check that `transaction()` was called with these particular ops or just `verify` that the call was made? Test is brittle because if implementation changes, it will fail.